lib/api/href.rb

```diff
@@ -1,34 +1,57 @@
 module Api
-  class HrefParser
-    def self.parse(href)
-      new(href).parse
-    end
-
+  class Href
     def initialize(href)
       @href = href
     end
 
-    def parse
-      return [nil, nil] unless href
-      if subcollection?
-        [subcollection.to_sym, ApplicationRecord.uncompress_id(subcollection_id)]
-      else
-        [collection.to_sym, ApplicationRecord.uncompress_id(collection_id)]
-      end
+    def resource
+      (subcollection? ? subcollection : collection)&.to_sym
     end
 
-    private
+    def resource_id
+      subcollection? ? subcollection_id : collection_id
+    end
 
-    attr_reader :href
+    def path
+      @path ||= remove_trailing_slashes(fully_qualified? ? URI.parse(href).path : ensure_prefix(href))
+    end
+
+    def collection
+      path_parts[version? ? 3 : 2]
+    end
+
+    def collection_id
+      ensure_uncompressed(path_parts[version? ? 4 : 3])
+    end
+
+    def subcollection
+      path_parts[version? ? 5 : 4]
+    end
+
+    def subcollection_id
+      ensure_uncompressed(path_parts[version? ? 6 : 5])
+    end
 
     def subcollection?
       !!subcollection
     end
 
-    def path
-      @path ||= remove_trailing_slashes(fully_qualified? ? URI.parse(href).path : ensure_prefix(href))
+    def version
+      @version ||= if version?
+                     path_parts[2][1..-1]
+                   else
+                     ApiConfig.base.version
```

Comment:
Is this wise? If `Href` can't deduce the version from the string, I'd figure it would give me `nil` or something, not just YOLO and feed me the default version from the application configuration. If the callers are few, maybe let the callers default there instead of in here?

Reply:
You are totally right. I got a bit carried away with the delegation. I think there is one caller, and I'm going to update it to do exactly as you suggested 😁